keps/20190218-cluster-resources-for-crds.md

```diff
@@ -0,0 +1,106 @@
+dependency from all objects.  Allowing a Framework or FrameworkVersion to define a set of Cluster objects that are present
+to support the creation and management of CRDs would circumvent the CRD from having to create and manage the object.
+
+
+## Table of Contents
+
+
+* [Table of Contents](#table-of-contents)
+* [Summary](#summary)
+* [Motivation](#motivation)
+    * [Goals](#goals)
+    * [Non-Goals](#non-goals)
+* [Proposal](#proposal)
+    * [User Stories [optional]](#user-stories-optional)
+      * [Story 1](#story-1)
+      * [Story 2](#story-2)
+    * [Implementation Details/Notes/Constraints [optional]](#implementation-detailsnotesconstraints-optional)
+    * [Risks and Mitigations](#risks-and-mitigations)
+* [Graduation Criteria](#graduation-criteria)
+* [Implementation History](#implementation-history)
+* [Drawbacks [optional]](#drawbacks-optional)
+* [Alternatives [optional]](#alternatives-optional)
+* [Infrastructure Needed [optional]](#infrastructure-needed-optional)
+
+[Tools for generating]: https://github.com/ekalinin/github-markdown-toc
+
+## Summary
+
+In order to enable the creation use of certain CRDs, some cluster scoped objects, or common objects, may need to be created.  This should not be
+part of the creation of a particular CRD instantiation since the deletion of that instance would remove the 
+dependency from all objects.  Allowing a Framework or FrameworkVersion to define a set of Cluster objects that are present
+to support the creation and management of CRDs would circumvent the CRD from having to create and manage the object.
+
+## Motivation
+
+### Goals
+
+* Allow the creation of common Kubernetes components in support of running a CRD.
+
+### Non-Goals
+
+* A mechanism for CRDs to create cluster scoped objects.
+
+## Proposal
+
+### User Stories [optional]
+
+
+#### Story 1
+
+A database Framework may want a Restic server deployed as part of the Framework to provide a central location for storing backups. 
+The capability here would define a namespace, service and deployment that provides Restic to the instance of the Framework to use
+by default.
+
+
+#### Story 2
+
+A Framework could leverage a MutatingWebHook for modifying pods deployments based on Node metadata (e.g. use a different image for nodes configured with GPU, PMEM, etc)
+
+
+#### Story 3
+
+The creation of a CRD should be controlled by a ClusterRole that defines permissions on who can create instances of the CRD.  These
+should probably be created
+
+#### Story 4
+
+FrameworkVersions require the existance of CRDs that are not controlled by Kudo (e.g. ETCD Operator) and require those to be installed when FV is enabled.
+
+### Implementation Details/Notes/Constraints [optional]
+
+
+
+### Risks and Mitigations
+
+
+
+## Graduation Criteria
+
+The MySQL Framework could be modified to deploy a central repo for backups and leverage those in each instance of MySQL
```

Comment:
what do you mean by `repo` here?

Reply:
For this conop, it was focused on a single place that all instances can use as a backup directory.  In the current MySQL Framework, each Instance has its own PVC that's used to store the backup for that Instance.  Rather than needing each to use its own, we could use a central location.